mon-pix/app/mixins/progression-tracker.js

```diff
@@ -0,0 +1,49 @@
+import EmberObject from '@ember/object';
+import { computed } from '@ember/object';
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  active: null,
+  steps: null,
+  stepsAhead: null,
+  stepsActivated: null,
+  progressions: null,
+
+  init() {
+    this._super(...arguments);
+    this.stepsAhead = [...this.steps].map(
+      (step) => EmberObject.create({ name: step, status: null })
+    );
+    this.stepsActivated = [];
+    this.next();
+  },
+  progression: computed('stepsAhead', 'stepActivated', function() {
+    return this.stepsActivated.concat(this.stepsAhead);
+  }),
+  next() {
+    if (!this.hasNext()) {
+      throw new Error('Tracker error, no more steps');
+    }
+    const nextStep = this.stepsAhead.shift();
+    nextStep.set('status', 'completed');
+    this.stepsActivated.push(nextStep);
+    this.set('active', nextStep.name);
+  },
+  prev() {
+    if (!this.hasPrevious()) {
+      throw new Error('Tracker error, no steps before');
+    }
+    const currentStep = this.stepsActivated.pop();
+    const previousStep = this.stepsActivated.pop();
+    currentStep.set('status', null);
+    this.set('stepsAhead', [currentStep].concat(this.stepsAhead));
```

Comment:
`[currentStep].concat(this.stepsAhead));` --> `[currentStep, ...this.stepsAhead]`

Reply:
Yes